spring-cloud-gcp-docs/src/main/asciidoc/pubsub.adoc

```diff
@@ -95,6 +97,13 @@ to be configured.
 
 `PubSubTemplate` uses a special subscriber generated by its `SubscriberFactory` to pull messages.
 
+If the message payload contains a serialized POJO, it can be retrieved as a `Class` compatible with that serialized payload:
+
+[source,java]
+----
+this.pubSubTemplate.getPayloadFromMessage(message, MyPojo.class);
```

Comment:
I believe the class information / default typing is not needed, if you are passing in the type.

Reply:
@meltsufin I tried getting the reader mapper without providing the class param, but it would not work. Jackson complains that it couldn't create its reader mapper object.

I infer from their  documentation (https://github.com/FasterXML/jackson-docs/wiki/JacksonPolymorphicDeserialization#11-global-default-typing) that this type information is not for the purpose of omitting this class param (which we can do if we did Java serialization), but for purposes of using super-types with serialized children.

Reply:
Is this supposed to work: `this.pubSubTemplate.getPayloadFromMessage(message, Object.class)`?
@artembilan mentioned that he will share some more details about how he has done this in the latest project.

Reply:
No, I promised to do that for the `HeaderMapper` PR #658 .

I will come back to you with review some time later. Don't know when though: the pool of tasks never exhausts...

Reply:
Sorry, my mistake. :)

Reply:
Right, @meltsufin , that should work:
See source code of the `GenericJackson2JsonRedisSerializer` where borrowed an idea about default typing:
```
public <T> T deserialize(@Nullable byte[] source, Class<T> type) throws SerializationException {

	Assert.notNull(type,
			"Deserialization type must not be null! Pleaes provide Object.class to make use of Jackson2 default typing.");

	if (SerializationUtils.isEmpty(source)) {
		return null;
	}

	try {
		return mapper.readValue(source, type);
	} catch (Exception ex) {
		throw new SerializationException("Could not read JSON: " + ex.getMessage(), ex);
	}
}
```
https://github.com/spring-projects/spring-data-redis/blob/master/src/main/java/org/springframework/data/redis/serializer/GenericJackson2JsonRedisSerializer.java